core/app/models/spree/store.rb

```diff
@@ -51,6 +51,18 @@ def unique_name
       "#{name} (#{code})"
     end
 
+    def countries_available_for_checkout
+      checkout_zone_or_default.try(:country_list) || Spree::Country.all
+    end
+
+    def states_available_for_checkout(country)
+      checkout_zone_or_default.try(:state_list_for, country) || country.states
+    end
+
+    def checkout_zone_or_default
```

Comment:
```suggestion
    def checkout_zone
```
in fact you will return here always `checkout_zone` and for other methods where you use that method it's not relevant if you return checkout zone set by user or default

Reply:
I think there is a a possibility where `default_zone` is returned:

- [add checkout zone filed to store](https://github.com/spree/spree/blob/master/core/db/migrate/20201006110150_add_checkout_zone_field_to_store.rb#L4) migration sets no not null constraint so any new store could be persisted with `checkout_zone-id = null`

Apart from that, `def checkout_zone` would be not a valid method name as it's already taken by the ActiveRecord relation

Reply:
right, forgot that I added that `checkout_zone` by myself 😂